src/library/scala/collection/View.scala

```diff
@@ -274,17 +303,21 @@ object View extends IterableFactory[View] {
       def hasNext: Boolean = it.hasNext
     }
     override def knownSize: Int = underlying.knownSize
+    override def isEmpty: Boolean = iterator.isEmpty
   }
 
   @SerialVersionUID(3L)
   private[collection] class Patched[A](underlying: SomeIterableOps[A], from: Int, other: IterableOnce[A], replaced: Int) extends AbstractView[A] {
     def iterator: Iterator[A] = underlying.iterator.patch(from, other.iterator, replaced)
+    override def knownSize: Int = if (underlying.knownSize == 0) 0 else super.knownSize
+    override def isEmpty: Boolean = iterator.isEmpty
```

Comment:
These 2 could take even more advantage of `underlying.knownSize` and `other.knownSize`.